src/types/topology/Node.types.ts

```diff
@@ -43,6 +43,8 @@ export interface TopologyAbstractNode extends TopologyComponent {
 
 export interface TopologyNode extends TopologyAbstractNode {
   // Properties
+  role: number;
```

Comment:
role and type are specific to ncd2 so they should not be defined here. We could extend TopologyNode in ncd2 to add those types for the `assignRoles` function instead.

https://www.logicbig.com/tutorials/misc/typescript/interface-extending-interfaces.html